src/tensor/cpu/allocate.rs

```diff
@@ -197,6 +197,29 @@ impl<E: Unit, const M: usize, const N: usize, const O: usize, const P: usize>
     }
 }
 
+impl<E: Unit> TensorFromVec<E> for Cpu {
+    fn try_tensor_from_vec<S: Shape>(
+        &self,
+        mut src: Vec<E>,
+        shape: S,
+    ) -> Result<Tensor<S, E, Self>, Self::Err> {
+        let num_elements = shape.num_elements();
+
+        if src.len() < num_elements {
```

Comment:
We should do this if there are too many elements as well - the truncation is a too silent in my opinion.